sequence_processing_pipeline/NuQCJob.py

```diff
@@ -0,0 +1,368 @@
+        lines.append("#SBATCH --mem 20gb\n")
+        lines.append("#SBATCH -N 1\n")
+        lines.append("#SBATCH -c 8\n")
+        lines.append("#SBATCH --output %x-%A_%a.out\n")
+        lines.append("#SBATCH --error %x-%A_%a.err\n")
+        lines.append("\n")
+        lines.append("if [[ -z \"${SLURM_ARRAY_TASK_ID}\" ]]; then\n")
+        lines.append("\techo \"Not operating within an array\"\n")
+        lines.append("\texit 1\n")
+        lines.append("fi\n")
+        lines.append("\n")
+        lines.append("mamba activate human-depletion\n")
+        lines.append("\n")
+        lines.append("set -x\n")
+        lines.append("set -e\n")
+        lines.append("\n")
+        lines.append("# set a temp directory, make a new unique one under it"
+                     " and\n")
+        lines.append("# make sure we clean up as we're dumping to shm\n")
+        lines.append("# DO NOT do this casually. Only do a clean up like this"
+                     " if\n")
+        lines.append("# you know for sure TMPDIR is what you want.\n")
+        lines.append("\n")
+        lines.append("mkdir -p {TMPDIR}\n")
+        lines.append("export TMPDIR=${TMPDIR}\n")
+        lines.append("export TMPDIR=$(mktemp -d)\n")
+        lines.append("echo $TMPDIR\n")
+        lines.append("\n")
+        lines.append("function cleanup {\n")
+        lines.append("\techo \"Removing $TMPDIR\"\n")
+        lines.append("\trm -fr $TMPDIR\n")
+        lines.append("\tunset TMPDIR\n")
+        lines.append("}\n")
+        lines.append("trap cleanup EXIT\n")
+        lines.append("\n")
+        lines.append("export FILES=$(pwd)/$(printf \"%s-%d\" ${PREFIX} "
+                     "${SLURM_ARRAY_TASK_ID})\n")
+        lines.append("if [[ ! -f ${FILES} ]]; then\n")
+        lines.append("\tlogger ${FILES} not found\n")
+        lines.append("\texit 1\n")
+        lines.append("fi\n")
+        lines.append("\n")
+        lines.append("delimiter=::MUX::\n")
+        lines.append("n=$(wc -l ${FILES} | cut -f 1 -d\" \")\n")
+        lines.append("\n")
+        lines.append("for i in $(seq 1 ${n})\n")
+        lines.append("do\n")
+        lines.append("\tline=$(head -n ${i} ${FILES} | tail -n 1)\n")
+        lines.append("\tr1=$(echo ${line} | cut -f 1 -d\" \")\n")
+        lines.append("\tr2=$(echo ${line} | cut -f 2 -d\" \")\n")
+        lines.append("\tbase=$(echo ${line} | cut -f 3 -d\" \")\n")
+        lines.append("\tr1_name=$(basename ${r1} .fastq.gz)\n")
+        lines.append("\tr2_name=$(basename ${r2} .fastq.gz)\n")
+        lines.append("\n")
+        lines.append("\techo \"${i}	${r1_name}	${r2_name}	${base}\" >> "
+                     "${TMPDIR}/id_map\n")
+        lines.append("\n")
+        lines.append("\tfastp \\\n")
+        lines.append("\t\t-l 45 \\\n")
+        lines.append("\t\t-i ${r1} \\\n")
+        lines.append("\t\t-I ${r2} \\\n")
+        lines.append("\t\t-w 7 \\\n")
+        lines.append("\t\t--adapter_fasta fastp_known_adapters_formatted.fna"
```

Comment:
This doesn't seem to be part of the PR?

Reply:
Not sure what you mean, but in any case I think the issue is resolved. I've replaced that method to generate the job script with calls using Jinja2 and the results match the original job script file.